<a href="https://colab.research.google.com/github/Rumas0/Thesis_work_SSL-imbalance/blob/main/Downloading_real_images.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Mounting Drive and loading images**

In [9]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [10]:
import pandas as pd
import os
import requests
from tqdm import tqdm
from google.colab import drive

In [11]:
backup_dir = '/content/drive/MyDrive/Thesis-work/Backups/thesis_backup_day1'

train_df = pd.read_csv(f'{backup_dir}/expA_train.csv')
val_df = pd.read_csv(f'{backup_dir}/expA_val.csv')
test_df = pd.read_csv(f'{backup_dir}/expA_test.csv')

all_images = pd.concat([train_df, val_df, test_df])['image'].unique()
print(f"Total unique images needed: {len(all_images)}")

os.makedirs('data/labeled_real', exist_ok=True)

Total unique images needed: 691


#**Kaggle Download**

**Setup Kaggle API**

In [7]:
from google.colab import files
print("Upload kaggle.json API token:")
files.upload()

!mkdir -p ~/.kaggle
!cp kaggle.json ~/.kaggle/
!chmod 600 ~/.kaggle/kaggle.json

!ls -ls ~/.kaggle/kaggle.json

Upload kaggle.json API token:


Saving kaggle.json to kaggle.json
4 -rw------- 1 root root 70 May 10 14:45 /root/.kaggle/kaggle.json


**Downloading ISIC 2019**

In [8]:
!kaggle datasets download -d bhanupratapbiswas/isic-2019

!unzip -q isic-2019.zip -d data/isic2019_full/
print("Download Completed")

403 Client Error: Forbidden for url: https://api.kaggle.com/v1/datasets.DatasetApiService/GetDatasetMetadata
unzip:  cannot find or open isic-2019.zip, isic-2019.zip.zip or isic-2019.zip.ZIP.
Download Completed


**Note: Kaggle API and ISIC direct download encountered 403 errors
 *# Real images obtained via manual download from ISIC Archive
 *# URL: https://challenge.isic-archive.com/data/
*# File: ISIC_2019_Training_Input.zip (~9GB)**